### PR TITLE
Unit Color Replacement via Config

### PR DESCRIPTION
**Pull request by mrschick**
I developed a rudimentary way to switch specific unit colours with others, by defining them in the config.

My main goal was simply to make BLUFOR units more visible, as I found the standard "Blue" (#0000FF) colour imported from Tacview to be too dark and hard to read at a glance.

But maybe in the future this could be developed further into some colour-blindness options 🤷🏻‍♂️

### CODE REVIEW

File: src/game_objects.py [hunk 1]
```diff
@@ -30,7 +33,16 @@ def __init__(self, object: ACMIObject, color: pygame.Color = pygame.Color(255,25
...
         self.font = font
         
-        self.color = pygame.Color(self.data.Color) 
+        self.color = pygame.Color(self.data.Color)
+
+        # Switch the object's color from its default to a respective replacement color
+        for i in config.app_config.get("map", "unit_color_switching", list):
+            print(self.color, "==", i[0])
+            if self.color == pygame.Color(i[0]):
+                if type(i[1]) != tomlkit.items.Array:
```
Comment (bobmoretti) [inline src/game_objects.py:42]: Typically, explicit checking for type equality is not the best way to solve whatever problem it's trying to solve.

At the very least, you could use `isinstance()`, that way subclasses will also evaluate to true.

I'm not familiar with tomlkit though. Is this how it's typically done or something?

Comment (mrschick) [reply src/game_objects.py]: No, it's dumber than that, I simply didn't know about `isinstance()` 🙃

Comment (bobmoretti) [reply src/game_objects.py]: No problem, that's the benefit of code review, we learn things from our peers!

**Comment by Briland5** [comment]
I'd like to eventually replace this with a feature that abstracts the idea of a Coalition and lets you override the color for the entire coalition at runtime, but this is good enough to include in the short term until that makes it in.

**Pull request merged**